book/src/slashing.md

```diff
@@ -0,0 +1,56 @@
+# Slashing
+
+Proof of Stake consensus requires that Byzantine behavior be punished by loss of stake and rewards, also known as slashing. 
+Solana implements slashing in 2 regimes: the validator regime and the replicator regime, corresponding to the two rewards regimes.
+During slashing, the network destroys some "penalty percentage" of lamports staked/earned in the rewards-generating accounts.
+
+## Validator regime
+
+Validators cast their votes to a Vote account to which stake has been [passively delegated](stake-delegation-and-rewards.md).
+The Vote program keeps track of rewards for all of the stakes, which allows it to be a central point for slashing.
+The basic mechanism of slashing is:
+
+1. a proof of Byzantine behavior is submitted to the Vote program
+2. the Vote program marks the Vote account as "slashed", which means:
+    * no further votes may be submitted to this account
```

Comment:
This punishment is worse then death :). Since validators have to get all their customers to redelegate.

Reply:
yes.  alternative?

once slashed, the account should probably hang for some amount of time, then be allowed to evaporate

none of the above is implemented

I suppose I can put a placeholder in for `slash()` and figure it out later.

Reply:
The signing key is marked invalid, account is frozen for an epoch?

Reply:
and slashing (i.e. sending TXs to all the stakes) can only be done while the account is frozen?

Reply:
Are we slashing stakes individually?

Reply:
we can batch 'em I suppose (limited number of accounts per TX), but someone has to pay for the TXes that mod the stakes